lib/utils/package_utils.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'set'
+
+module MeasureRepositoryServiceTestKit
+  # Utility functions in support of the $package test group
+  module PackageUtils
+    def related_artifacts_present?(bundle)
+      artifact_urls = Set[]
+      bundle.entry.each do |e|
+        next unless e.resource.resourceType == 'Library'
+
+        add_artifact_urls(artifact_urls, e.resource)
+      end
+      artifact_in_bundle?(artifact_urls, bundle)
+    end
+
+    def add_artifact_urls(artifact_urls, library)
+      library.relatedArtifact.each do |ra|
+        if ra.type == 'depends-on' && ra.resource.include?('Library') && ra.resource != 'http://fhir.org/guides/cqf/common/Library/FHIR-ModelInfo|4.0.1'
+          artifact_urls.add(ra.resource)
+        end
+      end
+    end
+
+    def artifact_in_bundle?(artifact_urls, bundle)
+      artifact_urls.all? do |url|
+        if url.include? '|'
+          split_reference = url.split('|')
+          url = split_reference[0]
+          version = split_reference[1]
+        end
+        bundle.entry.any? do |e|
+          e.resource.url == url && e.resource.version == version
+        end
+      end
+    end
+
+    def retrieve_measure_from_bundle(measure_iden, iden_type, bundle)
+      entry = bundle.entry.find do |e|
+        e.resource.resourceType == 'Measure' && e.resource.send(iden_type) == measure_iden
```

Comment:
Yeah this was a ruby rabbit hole for me. In this situation, send is used to dynamically retrieve the property with key: `iden_type`. I believe that send just iterates over the objects keys and returns the value the object has stored at the specified key. This allows the `retrieve_measure_from_bundle` function to work with any identifying property, rather than needing individual functions to access the measure by specific properties.